doc/HOWTO/howto_setupAndDeployCustom.md

```diff
@@ -0,0 +1,144 @@
+
+### Soroban CLI
+In order to interact with your docker image, you will need the [Soroban CLI][cli]. With `cargo`, all you need is:
+```sh
+cargo install --locked soroban-cli --features opt
+```
+
+note that this gets placed in `~/.cargo/bin`, which is why you want that as part of your `PATH`.
+
+### Custom Soroban Contracts
+
+<!--- TODO: Make a megacontract and link it here-->
+
+You can use the Soroban CLI to set up a blank project, and add your own contract code to it (you may also optionally include any of their example contracts).
+
+To set up a new _project_ (not contract!), type:
+```sh
+soroban contract init <DIR>
+```
+This will populate `<DIR>` (e.g. `.`) with some basic files, most notably a `Cargo.toml` config. Your custom contract should be added in `<DIR>/contracts/<CONTRACT>`.
+For example, for a contract `Foo`, you'd create:
+```
+<DIR>
+├── Cargo.toml
+└── contracts
+    └── Foo
+        ├── Cargo.toml
+        └── src
+            ├── lib.rs
+            └── test.rs
+```
+
+Note that `Cargo.toml` in `Foo` is _not_ the same `Cargo.toml` in `<DIR>`! The latter gets created automagically with `soroban contract init`, the former you have to add yourself. Here's an example of `Cargo.toml` in the contract folder:
+```
+dev_dependencies = { soroban-sdk = { workspace = true } }
+[package]
+name = "foo"
+version = "0.0.0"
+edition = "2021"
+publish = false
+
+[lib]
+crate-type = ["cdylib"]
+doctest = false
+
+[dependencies]
+soroban-sdk = { workspace = true }
+
+[dev-dependencies]
+soroban-sdk = { version = "20.3.2", features = ["testutils"] }
+```
+
+Write your contract code in `lib.rs`, and tests in `test.rs`.
+Make sure all your tests work before deploying (with print outputs):
+```sh
+cargo test -- --nocapture
+```
+Once they do, running
+```sh
+soroban contract build
+```
+will create a `.wasm` file from your Rust code in 
+```
+<DIR>/target/wasm32-unknown-unknown/release/<CONTRACT>.wasm
+```
+This is the file you will need to deploy to the network.
+
+### Connecting the Pieces
+We first need to make the Soroban CLI aware of the network running in Docker, by configuring
+```sh
+soroban config network add <NETWORK> \
+    --rpc-url "http://localhost:8000/soroban/rpc" \
+    --network-passphrase <PASSPHRASE>
+```
+where `<NETWORK>` is the alias you want to use (e.g. `net`/`testnet`/`standalone`/etc.) and `<PASSPHRASE>` is any string passphrase (e.g. the docs use `"Standalone Network ; February 2017"` or `"Test SDF Network ; September 2015"`, but it can be anything).
+
+Next, we create agents within this network. To do this, run
+```sh
+soroban keys generate <AGENT> --network <NETWORK>
+```
+followed by
+```sh
+soroban config identity fund <AGENT> --network <NETWORK>
+```
+The latter will show a warning, saying that `<AGENT>` already exists, but if you call this command before the previous one it fails to execute, and if you don't call it at all the next command fails, so just ignore the warning.
```

Comment:
afaict `soroban keys generate` fails with the funding error if it cannot make a connection to the node. Otherwise, the `config identity fund` is not needed.